### PR TITLE
EXP: For big cube, be a minimalist

### PR DESCRIPTION
**Pull request by pllim**
Companion to:

* https://github.com/spacetelescope/jdaviz/pull/2839

Seems a little more performant when using https://github.com/spacetelescope/jdaviz/pull/2827 instead of `main` with these changes, but slider still too laggy to be usable (but not as laggy as `main`). Subset creation at least possible (not possible on `main`).

[🐱](https://jira.stsci.edu/browse/JDAT-4397)

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/parsers.py [hunk 1]
```diff
@@ -249,7 +253,11 @@ def _parse_hdulist(app, hdulist, file_name=None,
         if hdu.name != 'PRIMARY' and 'PRIMARY' in hdulist:
             metadata[PRIHDR_KEY] = standardize_metadata(hdulist['PRIMARY'].header)
 
-        sc = _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, hdulist=hdulist)
+        if not is_big_cube:
+            sc = _return_spectrum_with_correct_units(
+                flux, wcs, metadata, data_type, hdulist=hdulist)
+        else:
+            sc = Spectrum1D(flux=flux, wcs=wcs, meta=metadata)
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/parsers.py:260]: will this also skip any necessary units stuff?  Can we just set/pass `hdulist=None` and still use `_return_spectrum_with_correct_units` or is that still too expensive?

**Review by kecnry — COMMENTED** [review]
If I understand the diff correctly, this skip parsing the hdulist if the cube is over 8 million spaxels?  What are the consequences of that?

File: jdaviz/configs/cubeviz/plugins/parsers.py (hunk 1 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/parsers.py:260]: `_return_spectrum_with_correct_units` creates a new `Spectrum1D` object. I have not looked into whether it does a copy or what. I was just trying some possible low hanging fruits to see if anything helps. Given the ticket is only 2 points, I didn't dig too deep here. There is a chance we do not have to disable this (and I don't think we want to disable if we can avoid it).

**Comment by pllim** [comment]
> this skip parsing the hdulist if the cube is over 8 million spaxels?

It skips parsing uncert and mask but flux should still be loaded.

**Comment by pllim** [comment]
My diff here is only exploratory. It didn't completely fix the lag in loading and slice. I tried profiling the code on `main` but Snakeviz crashed. I ran out of time so I didn't try Snakeviz here. See the ticket comments for a full report. Thanks!

**Comment by kecnry** [comment]
> My diff here is only exploratory.

Gotcha 👍

**Comment by pllim** [comment]
Follow up tickets created. This is no longer necessary.

**Pull request closed without merging**